autowiring/CoreObjectDescriptor.h

```diff
@@ -36,7 +37,11 @@ struct CoreObjectDescriptor {
     pFilter(autowiring::fast_pointer_cast<ExceptionFilter>(value)),
     pBoltBase(autowiring::fast_pointer_cast<BoltBase>(value)),
     receivesEvents(
-      [this]{
+      [this, value]{
+        // Because we manually added AutowiringEvents to the JunctionBoxManager, check here also
+        if (autowiring::fast_pointer_cast<AutowiringEvents>(value))
```

Comment:
This shouldn't be necessary, it means that `AutowiringEvents` isn't appearing in the event type registry.  @gtremper, any thoughts?